src/gc/gc.cpp

```diff
@@ -34233,6 +34236,20 @@ HRESULT GCHeap::Initialize()
             {
                 pmask &= smask;
 
+#ifdef FEATURE_PAL
+                // GetCurrentProcessAffinityMask can return pmask=0 and smask=0 on
+                // systems with more than 1 NUMA node. The pmask decides the
+                // number of GC heaps to be used and the processors they are
+                // affinitized with. So pmask is now set to reflect that 64
+                // processors are available to begin with. The actual processors in
+                // the system may be lower and are taken into account before
+                // finalizing the number of heaps.
+                if (!pmask)
+                {
+                    pmask = 0xFFFFFFFFFFFFFFFF;
```

Comment:
have you tested this on 32-bit? if so I am surprised you didn't get a compiler warning here.

Reply:
I just tried building for x86 and did not see any compiler warnings.

Reply:
When I tried this with the compiler explorer, it displayed a warning on x86: 
`[x86-64 clang 7.0.0 #1] warning: implicit conversion from 'unsigned long long' to 'uintptr_t' (aka 'unsigned int') changes value from 18446744073709551615 to 4294967295 [-Wconstant-conversion]`

@vkvenkat have you actually tried the x86 on Unix?

Reply:
I didn't see any warnings when I built for x86 on Ubuntu, but theoretically it should not work. So I will use the BIT64 & BIT32 macros to fix this.

Reply:
@vkvenkat you can use UINTPTR_MAX.

Reply:
The ARM CI legs are failing due to this:
`3:29: error: implicit conversion from 'unsigned long long' to 'uintptr_t' (aka 'unsigned int') changes value from 18446744073709551615 to 4294967295 [-Werror,-Wconstant-conversion]`

Reply:
Ran into some build errors with UINTPTR_MAX. But was able to use [SIZE_T_MAX](https://github.com/dotnet/coreclr/blob/master/src/inc/clrtypes.h#L117-L129) instead.

Reply:
Pushed this change & squashed commits.

Reply:
@janvorli Some OSX CI builds are failing: `Undefined symbols for architecture x86_64: "_GetCurrentProcessorNumberEx", referenced from CPUGroupInfo::CalculateCurrentProcessorNumber() in libutilcodestaticnohost.a(util.cpp.o)`.  Do we need to revert to using GetProcAddress calls with exports in mscorwks_unixexports.src rather than direct API calls to fix this?

Reply:
It should work. Let me give it a quick try on my local mac.

Reply:
Ok, I've found the problem. The libutilcodestaticnohost were missing coreclrpal library. This fixes the issue:
```diff
diff --git a/src/utilcode/staticnohost/CMakeLists.txt b/src/utilcode/staticnohost/CMakeLists.txt
index eea4d60785..e66a5de40d 100644
--- a/src/utilcode/staticnohost/CMakeLists.txt
+++ b/src/utilcode/staticnohost/CMakeLists.txt
@@ -8,5 +8,5 @@ endif(WIN32)
 add_library_clr(utilcodestaticnohost STATIC ${UTILCODE_STATICNOHOST_SOURCES})

 if(CLR_CMAKE_PLATFORM_UNIX)
-  target_link_libraries(utilcodestaticnohost  nativeresourcestring)
+  target_link_libraries(utilcodestaticnohost  nativeresourcestring coreclrpal)
 endif(CLR_CMAKE_PLATFORM_UNIX)

```

Reply:
Thanks, this fixes it for OSX. But I ran into this issue on Linux: `../../pal/src/libcoreclrpal.a(process.cpp.o): In function CreateProcessW: coreclr/src/pal/src/thread/process.cpp:530: multiple definition of CreateProcessW`
`CMakeFiles/mscordbi.dir/__/mscordac/palredefines.S.o:(.text+0x14f): first defined here`

Using a nested check for CLR_CMAKE_PLATFORM_DARWIN fixes this.